WEB-INF/classes/gov/noaa/pfel/erddap/handlers/EDDGridCopyHandler.java

```diff
@@ -0,0 +1,121 @@
+package gov.noaa.pfel.erddap.handlers;
+
+import static gov.noaa.pfel.erddap.dataset.EDDGrid.DEFAULT_MATCH_AXIS_N_DIGITS;
+
+import com.cohort.array.StringArray;
+import com.cohort.util.String2;
+import gov.noaa.pfel.erddap.dataset.EDD;
+import gov.noaa.pfel.erddap.dataset.EDDGrid;
+import gov.noaa.pfel.erddap.dataset.EDDGridCopy;
+import gov.noaa.pfel.erddap.util.EDStatic;
+import org.xml.sax.Attributes;
+import org.xml.sax.SAXException;
+
+public class EDDGridCopyHandler extends State {
+  private StringBuilder content = new StringBuilder();
+  private State completeState;
+  private String datasetID;
+  private SaxParsingContext context;
+
+  public EDDGridCopyHandler(
+      SaxHandler saxHandler, String datasetID, State completeState, SaxParsingContext context) {
+    super(saxHandler);
+    this.completeState = completeState;
+    this.datasetID = datasetID;
+    this.context = context;
+  }
+
+  private EDDGrid tSourceEdd = null;
+  private int tReloadEveryNMinutes = Integer.MAX_VALUE;
+  private String tAccessibleTo = null;
+  private String tGraphsAccessibleTo = null;
+  private boolean tAccessibleViaWMS = true;
+  private int tMatchAxisNDigits = DEFAULT_MATCH_AXIS_N_DIGITS;
+  private StringArray tOnChange = new StringArray();
+  private String tFgdcFile = null;
+  private String tIso19115File = null;
+  private boolean checkSourceData = true;
+  private boolean tFileTableInMemory = false;
+  private String tDefaultDataQuery = null;
+  private String tDefaultGraphQuery = null;
+  private int tnThreads = -1;
+  private boolean tAccessibleViaFiles = EDStatic.defaultAccessibleViaFiles;
+  private boolean tDimensionValuesInMemory = true;
+  private String tOnlySince = null;
+
+  @Override
+  public void startElement(String uri, String localName, String qName, Attributes attributes)
+      throws SAXException {
+    if (localName.equals("dataset")) {
```

Comment:
Should this throw an exception if it sees a second dataset?

Reply:
I don't think the old code specifies throwing an exception on getting a second child dataset,
https://github.com/ERDDAP/erddap/blob/main/WEB-INF/classes/gov/noaa/pfel/erddap/dataset/EDDGridFromEDDTable.java#L113-L127

Reply:
Yeah, add a check to make sure there's only one child dataset. Allowing multiple to be parsed could cause issues, and I think it's better to catch those early with explicit error messages.